### PR TITLE
Change Profile View

### PR DESCRIPTION
**Pull request by JustinLovesCompsci**
This must be merged after #53 
Added edit email/name/password in Profile View

For users with Facebook login, the Profile view looks like (without password edit):
screen shot 2015-04-26 at 6 02 00 pm

For users with email login, the profile view looks like (with password edit):
screen shot 2015-04-26 at 6 02 14 pm

### CODE REVIEW

File: SwiftParseChat/ProfileTableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if isLoginByFB() {
+            return actionItemsWithFB.count
+        } else {
+            return actionItemsWithoutFB.count
+        }
+    }
+    
+    override func tableView(tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return 0.01
```
Comment (huyouare) [inline SwiftParseChat/ProfileTableViewController.swift:44]: @JustinLovesCompsci Just curious why the 0.01 is necessary?

Comment (JustinLovesCompsci) [inline SwiftParseChat/ProfileTableViewController.swift:44]: If put it as 0.0, then you will see many empty lines. By setting 0.01, we have a nice border

**Comment by JustinLovesCompsci** [comment]
Refactored validateEmail according to @huyouare 's commits in another pull request #57 . 
Merge this before #57

File: SwiftParseChat/ProfileTableViewController.swift (hunk 1 above)
Comment (huyouare) [inline SwiftParseChat/ProfileTableViewController.swift:44]: Ah. Gotcha - thanks! Why does this work though?

Comment (JustinLovesCompsci) [inline SwiftParseChat/ProfileTableViewController.swift:44]: I guess by default ios would add empty lines...

**Comment by huyouare** [comment]
Actually, @JustinLovesCompsci why are we allowing users to change their emails? It should remain as more-or-less a primary key I think. Even allowing Facebook users to change names is a bit iffy.
I also want to add a phone number field. FYI I'm working on this branch right now

**Comment by JustinLovesCompsci** [comment]
I did not know email is a primary key. Um....well I am thinking people may need each other's email address to send over files/documents/codes for further collaboration. In ChatSettings, we can let user click on a person's image and go to the person's profile to get the email address (phone number). Facebook users do not have emails. This can be a way to let them add email. 

Another note: I think in general users should be able to change any info related to their profile....

**Comment by huyouare** [comment]
It's not really a primary key, I just think it should be used like one. One
-possible- case is if a user signed up with dummy account and wanted to
switch.
I wasn't thinking about allowing email addresses or phone numbers to be
shared (privacy). Also FB users all have the email fetched on registration.
I just fixed that. Let's just disable the edit email field for now, is that
okay?

**Comment by JustinLovesCompsci** [comment]
Yes that's fine. I was thinking about allowing users to set privacy, but this functionality may be a little bit too advanced for version 1.0.

**Comment by JustinLovesCompsci** [comment]
Anything else you need to change for this pull request? @huyouare

**Comment by huyouare** [comment]
Just going to disable for now so we can get the PR in. In the future, we should reconsider allowing the name to be changed and add a phone number field.

**Comment by JustinLovesCompsci** [comment]
Cool. merge now?

**Pull request merged**